code/go/internal/validator/folder_item_spec.go

```diff
@@ -16,20 +16,21 @@ import (
 	"github.com/xeipuuv/gojsonschema"
 
 	ve "github.com/elastic/package-spec/code/go/internal/errors"
+	"github.com/elastic/package-spec/code/go/internal/spectypes"
 	"github.com/elastic/package-spec/code/go/internal/validator/semantic"
 	"github.com/elastic/package-spec/code/go/internal/yamlschema"
 )
 
 type folderItemSpec struct {
-	Description       string   `yaml:"description"`
-	ItemType          string   `yaml:"type"`
-	ContentMediaType  string   `yaml:"contentMediaType"`
-	ForbiddenPatterns []string `yaml:"forbiddenPatterns"`
-	Name              string   `yaml:"name"`
-	Pattern           string   `yaml:"pattern"`
-	Required          bool     `yaml:"required"`
-	Ref               string   `yaml:"$ref"`
-	Visibility        string   `yaml:"visibility" default:"public"`
+	Description       string                 `yaml:"description"`
+	ItemType          string                 `yaml:"type"`
+	ContentMediaType  *spectypes.ContentType `yaml:"contentMediaType"`
```

Comment:
Why exactly do we need the `spectypes.ContentType`?

Reply:
I found the need of having the content type parsed in different places, then I decided to parse it directly when unmarshaling.

Reply:
ok, so the only thing we need to remember is to not load the file content if we expect to have packages weighing a few hundred of megabytes.

Reply:
Already done, this was one of my main motivations to refactor [`loadItemContent`](https://github.com/elastic/package-spec/pull/278/files#diff-533ad86ff7ed72649d73fc50169d7666aa1635da1376499b8cfbbf496be9ae03L73) and split its responsibilities.

Now the content of files is only load [when needed](https://github.com/elastic/package-spec/pull/278/files#diff-533ad86ff7ed72649d73fc50169d7666aa1635da1376499b8cfbbf496be9ae03R105), and after [checking the limits](https://github.com/elastic/package-spec/pull/278/files#diff-533ad86ff7ed72649d73fc50169d7666aa1635da1376499b8cfbbf496be9ae03R73-R86).

The only problem the new approach is that I am doing redundant `Stats`, but this should be much better that loading the content of all files, and file info could be reused in a further refactor. In principle I would leave further refactors there for a follow up.